### PR TITLE
sysinfo: Per test collectibles

### PR DESCRIPTION
**Pull request by ldoktor**
We have the support for it every since the beginning, let's allow it's usage as users asked for it https://github.com/avocado-framework/avocado/issues/2169

### CODE REVIEW

**Comment by narasimhan-v** [comment]
I do not see collectibles for per variant yet. Is that expected ?

```
# du -sh /root/sim/tests/results/latest/test-results/*/sysinfo/*128K	/root/sim/tests/results/latest/test-results/1-softwareraid.py_SoftwareRaid.test_run;raidlinear-scenario-3987/sysinfo/post
0	/root/sim/tests/results/latest/test-results/1-softwareraid.py_SoftwareRaid.test_run;raidlinear-scenario-3987/sysinfo/pre
0	/root/sim/tests/results/latest/test-results/1-softwareraid.py_SoftwareRaid.test_run;raidlinear-scenario-3987/sysinfo/profile
128K	/root/sim/tests/results/latest/test-results/2-softwareraid.py_SoftwareRaid.test_run;raid0-scenario-c9aa/sysinfo/post
0	/root/sim/tests/results/latest/test-results/2-softwareraid.py_SoftwareRaid.test_run;raid0-scenario-c9aa/sysinfo/pre
0	/root/sim/tests/results/latest/test-results/2-softwareraid.py_SoftwareRaid.test_run;raid0-scenario-c9aa/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/3-softwareraid.py_SoftwareRaid.test_run;raid1-scenario-4bb9/sysinfo/post
0	/root/sim/tests/results/latest/test-results/3-softwareraid.py_SoftwareRaid.test_run;raid1-scenario-4bb9/sysinfo/pre
0	/root/sim/tests/results/latest/test-results/3-softwareraid.py_SoftwareRaid.test_run;raid1-scenario-4bb9/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/4-softwareraid.py_SoftwareRaid.test_run;raid5-scenario-df88/sysinfo/post
0	/root/sim/tests/results/latest/test-results/4-softwareraid.py_SoftwareRaid.test_run;raid5-scenario-df88/sysinfo/pre
0	/root/sim/tests/results/latest/test-results/4-softwareraid.py_SoftwareRaid.test_run;raid5-scenario-df88/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/5-softwareraid.py_SoftwareRaid.test_run;raid10-scenario-e3f0/sysinfo/post
0	/root/sim/tests/results/latest/test-results/5-softwareraid.py_SoftwareRaid.test_run;raid10-scenario-e3f0/sysinfo/pre
0	/root/sim/tests/results/latest/test-results/5-softwareraid.py_SoftwareRaid.test_run;raid10-scenario-e3f0/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/6-softwareraid.py_SoftwareRaid.test_run;raid6-scenario-0ffd/sysinfo/post
0	/root/sim/tests/results/latest/test-results/6-softwareraid.py_SoftwareRaid.test_run;raid6-scenario-0ffd/sysinfo/pre
0	/root/sim/tests/results/latest/test-results/6-softwareraid.py_SoftwareRaid.test_run;raid6-scenario-0ffd/sysinfo/profile
```

**Comment by ldoktor** [comment]
Hello @narasimhan-v did you enable it in config? (in your `job.log` there should be `sysinfo.collectibles.collect_per_test    True` and for profilers `sysinfo.collect.profiler         True`.)

**Comment by narasimhan-v** [comment]
@ldoktor Right, I missed to set `sysinfo.collectibles.collect_per_test`.
Works fine now.

```
# du -sh /root/sim/tests/results/latest/test-results/*/sysinfo/*
2.3M	/root/sim/tests/results/latest/test-results/1-softwareraid.py_SoftwareRaid.test_run;raidlinear-scenario-3987/sysinfo/post
2.2M	/root/sim/tests/results/latest/test-results/1-softwareraid.py_SoftwareRaid.test_run;raidlinear-scenario-3987/sysinfo/pre
128K	/root/sim/tests/results/latest/test-results/1-softwareraid.py_SoftwareRaid.test_run;raidlinear-scenario-3987/sysinfo/profile
2.3M	/root/sim/tests/results/latest/test-results/2-softwareraid.py_SoftwareRaid.test_run;raid0-scenario-c9aa/sysinfo/post
2.2M	/root/sim/tests/results/latest/test-results/2-softwareraid.py_SoftwareRaid.test_run;raid0-scenario-c9aa/sysinfo/pre
128K	/root/sim/tests/results/latest/test-results/2-softwareraid.py_SoftwareRaid.test_run;raid0-scenario-c9aa/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/3-softwareraid.py_SoftwareRaid.test_run;raid1-scenario-4bb9/sysinfo/post
2.2M	/root/sim/tests/results/latest/test-results/3-softwareraid.py_SoftwareRaid.test_run;raid1-scenario-4bb9/sysinfo/pre
192K	/root/sim/tests/results/latest/test-results/3-softwareraid.py_SoftwareRaid.test_run;raid1-scenario-4bb9/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/4-softwareraid.py_SoftwareRaid.test_run;raid5-scenario-df88/sysinfo/post
2.2M	/root/sim/tests/results/latest/test-results/4-softwareraid.py_SoftwareRaid.test_run;raid5-scenario-df88/sysinfo/pre
192K	/root/sim/tests/results/latest/test-results/4-softwareraid.py_SoftwareRaid.test_run;raid5-scenario-df88/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/5-softwareraid.py_SoftwareRaid.test_run;raid10-scenario-e3f0/sysinfo/post
2.2M	/root/sim/tests/results/latest/test-results/5-softwareraid.py_SoftwareRaid.test_run;raid10-scenario-e3f0/sysinfo/pre
192K	/root/sim/tests/results/latest/test-results/5-softwareraid.py_SoftwareRaid.test_run;raid10-scenario-e3f0/sysinfo/profile
0	/root/sim/tests/results/latest/test-results/6-softwareraid.py_SoftwareRaid.test_run;raid6-scenario-0ffd/sysinfo/post
2.2M	/root/sim/tests/results/latest/test-results/6-softwareraid.py_SoftwareRaid.test_run;raid6-scenario-0ffd/sysinfo/pre
192K	/root/sim/tests/results/latest/test-results/6-softwareraid.py_SoftwareRaid.test_run;raid6-scenario-0ffd/sysinfo/profile
```

File: avocado/core/sysinfo.py [hunk 1]
```diff
@@ -601,7 +601,10 @@ def start_test_hook(self):
         Logging hook called before a test starts.
         """
         for log in self.start_test_collectibles:
-            log.run(self.pre_dir)
+            if isinstance(log, Daemon):  # log daemons in profile directory
```
Comment (clebergnu) [inline avocado/core/sysinfo.py]: Nice catch.

**Pull request closed without merging**